src/apps/frontend/pages/authentication/phone-login/phone-login-form.hook.ts

```diff
@@ -0,0 +1,65 @@
+import { useFormik } from 'formik';
+import { PhoneNumberUtil } from 'google-libphonenumber';
+import { useNavigate } from 'react-router-dom';
+import * as Yup from 'yup';
+
+import constant from '../../../constants';
+import constants from '../../../constants/routes';
+import { useAuthContext } from '../../../contexts';
+import { AsyncError } from '../../../types';
+
+interface PhoneLoginFormProps {
+  onSuccess: () => void;
+  onError: (err: AsyncError) => void;
+}
+const usePhoneLoginForm = ({ onSuccess, onError }: PhoneLoginFormProps) => {
+  const {
+    isSendOTPLoading, sendOTPError, sendOTP,
+  } = useAuthContext();
+
+  const navigate = useNavigate();
+
+  const formik = useFormik({
+    initialValues: {
+      countryCode: '+1',
+      country: 'US',
+      phoneNumber: '',
+    },
+    validationSchema: Yup.object({
+      phoneNumber: Yup.string()
+        .min(8, constant.PHONE_VALIDATION_ERROR)
+        .max(14, constant.PHONE_VALIDATION_ERROR)
```

Comment:
initialise these min and max numbers in constant file and use them here. Or directly include the length of the phone number in `countryCode` object of `COUNTRIES` array in src/apps/frontend/constants/countries.ts file and validate accordingly

Reply:
We don't want to get into validation of phone numbers, the rules can get complicated internationally. Can we not use Google phone lib here?

Reply:
done, yes we use Google phone lib.